File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        frequencyMap = defaultdict(int)
        for n in nums:
            frequencyMap[n] += 1
        res = []
        while k > 0:
            s = max(frequencyMap, key=frequencyMap.get)
            res.append(s)
            frequencyMap.pop(s, None)
            k -= 1
        return res

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<int, int> m;
        int l = 0, r = 0, maxf = 0;
        int res = 0;
        while (r < s.size()){
            m[s[r]]++;
            maxf = max(maxf, m[s[r]]);

            if ((r-l+1) - maxf > k){
                m[s[l]]--;
                l++;
            }
            res = max(r-l+1, res);
            r++;
            
        }
        return res;
    }
};
